scib/metrics/pcr.py

```diff
@@ -216,18 +271,39 @@ def pc_regression(
     else:
         if verbose:
             print("one-hot encode categorical values")
-        covariate = pd.get_dummies(covariate)
+        covariate = pd.get_dummies(covariate).to_numpy()
 
     # fit linear model for n_comps PCs
-    r2 = []
-    for i in range(n_comps):
-        pc = X_pca[:, [i]]
-        lm = LinearRegression()
-        lm.fit(covariate, pc)
-        r2_score = np.maximum(0, lm.score(covariate, pc))
-        r2.append(r2_score)
+    if verbose:
+        print(f"Use {n_threads} threads for regression...")
+    if n_threads == 1:
+        r2 = []
+        for i in tqdm(range(n_comps), total=n_comps):
+            r2_score = linreg_method(X=covariate, y=X_pca[:, [i]])
+            r2.append(np.maximum(0, r2_score))
+    else:
+        with ThreadPoolExecutor(max_workers=n_threads) as executor:
+            run_r2 = executor.map(linreg_method, [covariate] * n_comps, X_pca.T)
+            r2 = list(tqdm(run_r2, total=n_comps))
 
     Var = pca_var / sum(pca_var) * 100
     R2Var = sum(r2 * Var) / 100
 
     return R2Var
+
+
+def linreg_sklearn(X, y):
+    from sklearn.linear_model import LinearRegression
+
+    lm = LinearRegression()
+    lm.fit(X, y)
+    r2_score = lm.score(X, y)
+    np.maximum(0, r2_score)
+    return r2_score
+
+
+def linreg_np(X, y):
```

Comment:
I wonder if for PC regression we can do it even more efficiently. According to chatGPT, both `sklearn.linear_model.LinearRegression` and `numpy.lstsq` apply SVD to `X`. PCA, however, does the same (and additionally subsetting the components), so we don't need to recalculate SVD twice. Check this chat for more info (I am not sure that the information from it is 100% correct though): https://chat.openai.com/share/f14c210f-6620-4fba-9047-57fdccb55955

Reply:
I see, so you mean doing an SVD/PCA once and then computing the pseudo-inverse for each covariate? That's a cool idea! However, I think I'll keep it as is (i.e. minimal coding) for now and revisit it in another PR.

Reply:
Exactly! Yeah, this will likely not save you too much time, but it would make the code more complex. Maybe a cool thing to try later with proper testing that it works correctly